### PR TITLE
docs: Adds documentation for Modal component

### PR DESCRIPTION
**Pull request by zero0Halo**
docs: Created new components to assemble the pieces of the modal as an example

fix: Made the code block live for icon.mdx

docs: added the usage of the modal components to modal.mdx

docs: Added api for Modal

chore: updated anchor to 0.0.1-beta.8 for doc site

feat: added a StyledHr component to mdx's custom component bindings

feat: added a slight bottom margin to mdx custom h6 tag

**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config
- [ ] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

Adds documentation for the modal component. This includes components to help illustrate its use and detailed code blocks. No interactive code editor here, it didn't make sense with how Modal is designed.

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "author": "Kameron (@sharkparty)",
     "dependencies": {
         "@reach/component-component": "^0.1.3",
-        "@retailmenot/anchor": "^0.0.1-alpha.31",
+        "@retailmenot/anchor": "^0.0.1-beta.8",
```
Comment (bjohnsonrmn) [inline docs/package.json:7]: Just curious, what does `beta.8` mean?

File: docs/src/components/ModalExample/ModalExample.component.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    putting them at the top of the gatsby site like I did in the code examples. I may revist this
+    choice down the road, but for now I don't want to put the providers that high up in the application tree (yet).
+*/
+
+import * as React from 'react';
+import styled, { ThemeProvider } from 'styled-components';
+import { RootTheme, Typography, ModalProvider } from '@retailmenot/anchor';
+import { ModalAndButton } from './ModalAndButton.component';
+
+const StyledModalExample = styled.div`
```
Comment (bjohnsonrmn) [inline docs/src/components/ModalExample/ModalExample.component.tsx:15]: Could any of these styles be applied via Modal's props?

File: docs/src/pages/components/icon.mdx [hunk 3]
```diff
@@ -21,7 +21,7 @@ There are currently <IconCount /> icons available. Note that you can try any of
 
 #### Live Code Editor
 
-```tsx
+```tsx live
```
Comment (bjohnsonrmn) [inline docs/src/pages/components/icon.mdx:24]: I may have missed this in a different PR, but do we document what `live` does somewhere?

**Review by bjohnsonrmn — COMMENTED** [review]
I haven't finished going through all of this yet, but I figured I would go ahead and leave a few comments. Thanks for pioneering how to document components that require more setup. 

Did you ever see an error similar to this when building the doc site locally?
```
Error in "/Users/bjohnson/projects/anchor/docs/node_modules/gatsby-plugin-manifest/gatsby-node.js": Cannot find module 'gatsby/dist/utils/create-content-digest'
```

I'm seeing that after fresh dependency updates in both root and `docs/` 🤔 

EDIT:  this fixed it for me `npm i gatsby-plugin-manifest` ([more context](https://github.com/gatsbyjs/gatsby/issues/16096))

File: docs/package.json (hunk 1 above)
Comment (sharkparty) [reply docs/package.json:7]: We're still shy of a full release so we're using alpha > beta > rc as an indicator of when we're open for business.

File: docs/src/components/ModalExample/ModalExample.component.tsx (hunk 2 above)
Comment (sharkparty) [reply docs/src/components/ModalExample/ModalExample.component.tsx:15]: I wouldn't mind seeing a top-level theme provider injected into gatsby as well as seeing how much garbage collection that creates from unneeded, redundant style rules. What are your thoughts?

**Review by sharkparty — APPROVED** [review]
LGTM, but I would like to know what you think about elevating the theme provider to the root of the gatsby app.

File: docs/src/pages/components/icon.mdx (hunk 3 above)
Comment (zero0Halo) [reply docs/src/pages/components/icon.mdx:24]: No, but I have been thinking about putting together a style guide for documentation and including an explanation on that. For now it's word of mouth. I can add a comment and example use to the code editor component for the time being.

**Comment by zero0Halo** [comment]
Totally willing to try moving the ThemeProvider higher up and see what happens.

File: docs/src/pages/components/icon.mdx (hunk 3 above)
Comment (sharkparty) [reply docs/src/pages/components/icon.mdx:24]: I'm content adding a comment in the `Page` component (I think that's where that lives) calling this out and calling that documented.

**Comment by sharkparty** [comment]
Let's timebox that to EOB today and see where we get.

File: docs/src/components/ModalExample/ModalExample.component.tsx (hunk 2 above)
Comment (zero0Halo) [reply docs/src/components/ModalExample/ModalExample.component.tsx:15]: @bjohnson No, those styles are just for the look of the 'page'. They don't actually have anything to do with the modal's appearance or the button.

File: docs/src/pages/components/icon.mdx (hunk 3 above)
Comment (bjohnsonrmn) [reply docs/src/pages/components/icon.mdx:24]: Sounds good! Thanks for clarifying 👍

Comment (zero0Halo) [reply docs/src/pages/components/icon.mdx:24]: Added a fairly detailed example at the top of the `CodePreview` component showing use cases.

**Comment by zero0Halo** [comment]
As it turns out `ThemeProvider` and `RootTheme` were already high up in app tree in the `Page` component. I ended up removing them from `ModalExample`.

**Comment by zero0Halo** [comment]
@bjohnsonrmn I didn't get errors when running `gatsby develop` or `gatsby build` in either my current Anchor repo or a fresh install with `yarn`.

**Comment by brandonlilly** [comment]
Looks great. The only thing that I'd add is a blurb on `Content` behavior, specifically- If `Content` is on it's own it had padding on all sides. If you add a `Header` before it, it loses its top padding. If you add any siblings after it, it loses it bottom padding.

**Pull request merged**